src/cloudWatchLogs/changeLogSearch.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as telemetry from '../shared/telemetry/telemetry'
+import * as vscode from 'vscode'
+import { showInputBox } from '../shared/ui/inputPrompter'
+import { isLogStream } from './cloudWatchLogsUtils'
+import { prepareDocument } from './commands/searchLogGroup'
+import { getActiveUri } from './document/logStreamDocumentProvider'
+import { CloudWatchLogsData, filterLogEventsFromUriComponents, LogStreamRegistry } from './registry/logStreamRegistry'
+import { isViewAllEvents, TimeFilterResponse, TimeFilterSubmenu } from './timeFilterSubmenu'
+
+type ChangeableParam = 'filterPattern' | 'timeFilter'
+
+export async function getNewData(
+    registry: LogStreamRegistry,
+    param: ChangeableParam,
```

Comment:
very small nit:
```suggestion
    param: 'filterPattern' | 'timeFilter',
```

Not everything needs a name